pkg/apiserver/profiling/model.go

```diff
@@ -109,13 +129,14 @@ type TaskGroup struct {
 }
 
 // NewTaskGroup create a new profiling task group.
-func NewTaskGroup(db *dbstore.DB, profileDurationSecs uint, stats model.RequestTargetStatistics) *TaskGroup {
+func NewTaskGroup(db *dbstore.DB, profileDurationSecs uint, stats model.RequestTargetStatistics, profilingTypeList string) *TaskGroup {
```

Comment:
Maybe you can use this to make your life easier:

```go
type ProfilingTypeList []TaskProfilingType

func (r *ProfilingTypeList) Scan(src interface{}) error {
	return json.Unmarshal([]byte(src.(string)), r)
}

func (r ProfilingTypeList) Value() (driver.Value, error) {
	val, err := json.Marshal(r)
	return string(val), err
}

// ...

type TaskGroupModel struct {
	...
	RequestedProfilingTypes  ProfilingTypeList `json:"requested_profiling_types"`
	...
}
```

It relies on customized serialization / deserialization so that your model is kept to be structured.